src/SA/ldapsearch/entry.c

```diff
@@ -55,6 +56,52 @@ typedef PLDAPSearch LDAPAPI (*ldap_search_init_pageA_t)(PLDAP ExternalHandle,con
 #define WLDAP32$ldap_next_attribute ((ldap_next_attribute_t)DynamicLoad("WLDAP32$ldap_next_attribute"))
 #define WLDAP32$ldap_search_init_pageA ((ldap_search_init_pageA_t)DynamicLoad("WLDAP32$ldap_search_init_pageA"))
 
+LDAPControlA *FormatSDFlags(int iFlagValue)
+{
+  BerElement *pber = NULL;
+  PLDAPControlA pLControl = NULL;
+  PBERVAL pldctrl_value = NULL;
+  int success = -1;
+
+  internal_printf("%s", LDAP_SERVER_SD_FLAGS_OID);
+  // Format and encode the SEQUENCE data in a BerElement.
+
+  pber = WLDAP32$ber_alloc_t(LBER_USE_DER);
+  if(pber==NULL) return NULL;
+  pLControl = MSVCRT$calloc(1, sizeof(LDAPControlA));
+  if(pLControl==NULL) { WLDAP32$ber_free(pber,1); return NULL; }
+  WLDAP32$ber_printf(pber, (PSTR)"{i}", iFlagValue);
+
+  // Transfer the encoded data into a BERVAL.
+  success = WLDAP32$ber_flatten(pber, &pldctrl_value);
+  if(success == 0)
+  {
+      WLDAP32$ber_free(pber, 1);
+  }
+  else
+  {
+      internal_printf("ber_flatten failed");
+      // Call error handler here.
```

Comment:
As the comment states, error handling is missing here. You could still return NULL then handle that in the calling function